__tests__/app.test.js

```diff
@@ -187,3 +188,81 @@ expect(comments).toBeSortedBy('created_at',  {
         });
 
     })
+
+    describe('7. POST /api/reviews/:review_id/comments', () => {
+        test('should respond with a comment with username and body added ', () => {
+            const newComment = {
+                username: "bainesface",
+                body: 'this is a new comment'
+              }
+            return request(app)
+            .post("/api/reviews/2/comments")
+            .send(newComment)
+            .expect(201)
+            .then(({body})=>{
+            const {comment} = body
+            expect(comment).toEqual(
+                expect.objectContaining({
+                body: 'this is a new comment',
+                votes: 0,
+                author: "bainesface",
+                review_id: 2,
+                created_at: expect.any(String),
+              })
+            )
+            })
+        
+        });
+        test('should return status 400 when given a bad body', () => {
+            const invalidBody = {
+                username: "bainesface",
+                nigel: 'this is bad request'
+              }
+              
+            return request(app)
+            .post("/api/reviews/2/comments")
+            .send(invalidBody)
+            .expect(400)
+            .then((({body:{msg}})=>{
+                expect(msg).toBe('Bad Request')
+                
+                    }))
+
+                    
+        });
+        test('should return status 400 when given a bad username', () => {
+          
+              const invalidUsername = {
+                manyThings: "bainesface",
+                body: 'this is bad request'
+              }
+            return request(app)
+            .post("/api/reviews/2/comments")
+            .send(invalidUsername)
+            .expect(400)
+            .then((({body:{msg}})=>{
+                expect(msg).toBe('Bad Request')
+                
+                    }))
+
+                    
+        });
+        test('should respond with a 404 when given a bad review id', () => {
+            
+            const invalidUsername = {
+                username: "bainesface",
+                body: 'this is bad request'
+              }
+            return request(app)
+            .post("/api/reviews/2632/comments")
+            .send(invalidUsername)
+            .expect(404)
+            .then((({body:{msg}})=>{
+                expect(msg).toBe('Bad Request')
+                
+                    }))
+        });
+            
+        });
+    
+
```

Comment:
You should add a 404 test for a correctly formatted request but one where the user does not exist. This will throw a psql error which you may already be handling.